File: source_funcs/go_transcoder/CHECK_IF_A_NUMBER_IS_JUMBLED_OR_NOT/CHECK_IF_A_NUMBER_IS_JUMBLED_OR_NOT.go
package main

import "math"

func f_gold(num int) bool {
	if num/10 == 0 {
		return true
	}
	for num != 0 {
		if num/10 == 0 {
			return true
		}
		var digit1 int = num % 10
		var digit2 int = (num / 10) % 10
		if math.Abs(float64(digit2-digit1)) > 1 {
			return false
		}
		num = num / 10
	}
	return true
}

File: rust_funcs/go_transcoder/CHECK_IF_A_NUMBER_IS_JUMBLED_OR_NOT/CHECK_IF_A_NUMBER_IS_JUMBLED_OR_NOT.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(mut num: i32) -> i32 {
    if num / 10 == 0 {
        return 1;
    } 
    while num != 0 {
        let digit1 = num % 10;
        let digit2 = (num / 10) % 10;
        if (digit2 as f32 - digit1 as f32).abs() > 1.0 {
            return 0;
        }
        num = num / 10;
    }
    1
}
